src/ui/EditorNS/editor.cpp

```diff
@@ -48,6 +50,12 @@ namespace EditorNS
         query.addQueryItem("themePath", theme.path);
         query.addQueryItem("themeName", theme.name);
 
+        if (showLineNumbers) {
```

Comment:
Instead of passing the `showLineNumbers` item through the constructor, we can do it the way we apply all other items to an Editor.

Take a look at [this function](https://github.com/notepadqq/notepadqq/blob/master/src/ui/mainwindow.cpp#L1194). All new editors have this function called, and in there all settings are applied. We can have a `setLineNumbersVisible` in there as well.

The `theme` item is the odd one out. For some reason it's passed through the constructor.

Reply:
Okay I think I got it! I will remove my changes from the constructor, and use the function you referenced to make the option change. To be honest I was a bit confused from the use of the theme item, but now I understand the logic.

Reply:
@JuBan1 I think I did it! I look forward to your opinion/review.